core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -335,6 +352,52 @@ public boolean isSamplingEnabled() {
     public SamplerConfiguration getSamplerConfiguration() {
       return RFileScanner.this.getSamplerConfiguration();
     }
+
+    @Override
+    public TableId getTableId() {
+      return null;
+    }
+
+    @Override
+    @Deprecated(since = "2.0.0")
+    public AccumuloConfiguration getConfig() {
+      return tableConf;
+    }
+
+    @Override
+    @Deprecated(since = "2.1.0")
+    public ServiceEnvironment getServiceEnv() {
+      return serviceEnvironment.get();
+    }
+
+    private ServiceEnvironment createServiceEnv() {
+      return new ServiceEnvironment() {
+        @Override
+        public <T> T instantiate(TableId tableId, String className, Class<T> base) {
+          throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public <T> T instantiate(String className, Class<T> base) {
+          throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public String getTableName(TableId tableId) {
+          throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Configuration getConfiguration(TableId tableId) {
```

Comment:
It would be nice to update the javadoc for the following methods.

https://github.com/apache/accumulo/blob/fd111a22a6454837ad6076d5ba6fe54fc8934be5/core/src/main/java/org/apache/accumulo/core/client/rfile/RFile.java#L178-L187

To state that the iterator env will return the properties passed into these methods.  Could also include the version number in the docs.  Like `Starting with 2.1.4 <link to iter env method> will return the properties passed in here`

Reply:
Added and fixed an issue I noticed with a couple RFile javadocs